### ISSUE TITLE
Add reproducible tutorials comparing Infomap with Leiden and Louvain

### ISSUE THREAD

**Issue by antoneri** [mapequation/infomap#539]
Users often discover community detection methods through tutorials and benchmark-style examples. Leiden and Louvain are widely used defaults in many graph clustering workflows, especially in NetworkX, igraph, and Scanpy contexts.

Infomap adoption would benefit from practical, reproducible tutorials showing how to use Infomap as a drop-in or near-drop-in alternative, how outputs differ, and how to evaluate or visualize the result.

The goal is not to claim that one method is universally better. The goal is to help users understand how to run Infomap, compare results, and integrate it into existing pipelines.

## Goal

Create tutorial notebooks and documentation examples that compare Infomap with Leiden and Louvain across common workflows.

## Deliverables

Add at least three tutorials:

```text
docs/tutorials/compare_infomap_louvain_leiden_networkx.ipynb
docs/tutorials/compare_infomap_louvain_leiden_igraph.ipynb
docs/tutorials/infomap_scanpy_workflow.ipynb
```

If the documentation system does not use notebooks, equivalent Markdown pages with executable code blocks are acceptable.

## Tutorial 1: NetworkX workflow

Use a small built-in graph, preferably:

```python
nx.karate_club_graph()
```

Show:

1. load graph,
2. run Infomap,
3. run Louvain using NetworkX or a common optional package,
4. compare number of communities,
5. compare assignments in a pandas DataFrame,
6. visualize communities,
7. export to GraphML/GEXF with Infomap attributes if the export feature is available.

Example structure:

```python
import networkx as nx
import pandas as pd
from infomap import Infomap

G = nx.karate_club_graph()

im = Infomap.from_networkx(G)
im.run()
infomap_modules = im.get_modules()

df = pd.DataFrame({
    "node": list(G.nodes()),
    "infomap": [infomap_modules[n] for n in G.nodes()],
})
```

If Leiden is not naturally available in NetworkX, explain that Leiden is shown in the igraph tutorial.

## Tutorial 2: igraph workflow

Use an igraph-compatible graph.

Show:

1. load or construct graph,
2. run Infomap through the existing Infomap integration,
3. run igraph/community Louvain if available,
4. run Leiden if the optional `leidenalg` dependency is installed,
5. compare assignments,
6. visualize or summarize results.

Optional dependencies must be handled gracefully:

```python
try:
    import leidenalg
except ImportError:
    leidenalg = None
```

If `leidenalg` is missing, skip Leiden cells with a clear message rather than failing.

## Tutorial 3: Scanpy / AnnData workflow

Use a small built-in or synthetic AnnData dataset.

Show:

1. load AnnData,
2. build neighbors graph,
3. run Leiden using Scanpy if available,
4. run Infomap using the new AnnData integration,
5. compare cluster labels,
6. plot UMAP colored by Infomap and Leiden if Scanpy plotting is available.

Example:

```python
import scanpy as sc
import infomap

sc.pp.neighbors(adata)
sc.tl.umap(adata)

sc.tl.leiden(adata, key_added="leiden")
infomap.tl.infomap(adata, key_added="infomap")

sc.pl.umap(adata, color=["leiden", "infomap"])
```

## Evaluation metrics

At least one tutorial should include simple comparison metrics.

Preferred metrics:

```python
from sklearn.metrics import adjusted_rand_score, normalized_mutual_info_score

ari = adjusted_rand_score(df["infomap"], df["louvain"])
nmi = normalized_mutual_info_score(df["infomap"], df["louvain"])
```

If scikit-learn is unavailable, skip metrics with a clear note.

## Documentation requirements

Each tutorial should include:

- what the graph represents,
- what Infomap optimizes at a high level,
- how Infomap output differs from modularity-based methods,
- how to interpret module IDs,
- how to handle directed graphs,
- how to handle undirected graphs,
- how to handle weighted graphs,
- how to export results,
- how to cite Infomap.

Avoid marketing claims. Use neutral wording.

## Reproducibility requirements

Each tutorial must:

- run from a clean checkout,
- pin or document optional dependencies,
- use small datasets suitable for CI or docs builds,
- avoid network downloads unless cached or optional,
- set random seeds where relevant,
- state package versions in the notebook output or documentation.

Recommended helper cell:

```python
import infomap
import networkx as nx
import pandas as pd

print("infomap:", infomap.__version__)
print("networkx:", nx.__version__)
print("pandas:", pd.__version__)
```

## CI / documentation build

Add one of the following:

### Option A

Execute notebooks in CI.

### Option B

Add a lightweight smoke test that runs the main code paths without rendering plots.

### Option C

Mark notebooks as documentation-only but add separate Python scripts under:

```text
examples/
```

that are tested in CI.

Preferred example scripts:

```text
examples/compare_networkx.py
examples/compare_igraph.py
examples/scanpy_infomap.py
```

At minimum, add a CI smoke test for the NetworkX example.

## Files to add or update

Likely files:

```text
docs/tutorials/compare_infomap_louvain_leiden_networkx.ipynb
docs/tutorials/compare_infomap_louvain_leiden_igraph.ipynb
docs/tutorials/infomap_scanpy_workflow.ipynb
examples/compare_networkx.py
examples/compare_igraph.py
examples/scanpy_infomap.py
docs/index.md
docs/tutorials/index.md
pyproject.toml
docs requirements file
```

Adjust paths to match the existing repository structure.

## Acceptance criteria

- At least three tutorials are added or equivalent docs pages are created.
- Tutorials cover NetworkX, igraph, and AnnData/Scanpy workflows.
- Optional dependencies are handled gracefully.
- At least one tutorial computes ARI or NMI between methods.
- At least one tutorial shows visualization.
- At least one tutorial shows export or prepares data for export.
- A lightweight CI or docs smoke test verifies that the main examples run.
- Documentation links to the tutorials from the main docs navigation.
